hack/actions/gen/main.go

```diff
@@ -0,0 +1,689 @@
+		"make cmake/install",
+		"make buf/install",
+		"make hdf5/install",
+		"make helm-docs/install",
+		"make helm/install",
+		"make k3d/install",
+		"make k9s/install",
+		"make kind/install",
+		"make kubectl/install",
+		"make kubelinter/install",
+		"make reviewdog/install",
+		"make tparse/install",
+		"make valdcli/install",
+		"make yq/install",
+		"make minikube/install",
+		"make stern/install",
+		"make telepresence/install",
+	}
+
+	devContainerPreprocess = []string{
+		"curl -fsSL https://deb.nodesource.com/setup_current.x | bash -",
+		"apt-get clean",
+		"apt-get update -y",
+		"apt-get upgrade -y",
+		"apt-get install -y --no-install-recommends --fix-missing nodejs",
+		"npm install -g npm@latest",
+		"apt-get clean",
+		"apt-get autoclean -y",
+		"apt-get autoremove -y",
+		"make delve/install",
+		"make gomodifytags/install",
+		"make gopls/install",
+		"make gotests/install",
+		"make impl/install",
+		"make staticcheck/install",
+	}
+)
+
+func generateWorkflowStruct(data *Data) Workflow {
+	var workflow Workflow
+
+	workflow.Name = data.Name
+	workflow.On.Push.Branches = data.PushBranches
+	workflow.On.Push.Tags = data.PushTags
+	workflow.On.PullRequest.Paths = data.PullRequestPaths
+	workflow.On.PullRequestTarget.Paths = data.PullRequestTargetPaths
+	workflow.Jobs.Build.Uses = data.BuildUses
+	workflow.Jobs.Build.With.Target = data.BuildTarget
+	workflow.Jobs.Build.Secrets = data.BuildSecrets
+
+	if data.Cron != "" {
+		workflow.On.Schedule.Cron = data.Cron
+	}
+	if data.BuildPlatforms != "" {
+		workflow.Jobs.Build.With.Platforms = data.BuildPlatforms
+	}
+
+	return workflow
+}
+
+func appendComponentsPaths(paths []string, data *Data) []string {
+	paths = append(paths, "cmd/"+data.PackageDir+"/**", "pkg/"+data.PackageDir+"/**")
+
+	// append components' internals
+	if data.Name == "agent-faiss" || data.Name == "agent-ngt" {
+		paths = append(paths, "pkg/agent/internal/**")
+	}
+	if strings.Split(data.Name, "-")[0] == "gateway" {
+		paths = append(paths, "pkg/gateway/internal/**")
+	}
+
+	return paths
+}
+
+func appendContainerTypeDependingPaths(paths []string, data *Data) []string {
+	ct := data.ContainerType
+	if ct == DevContainer || ct == CIContainer {
+		paths = append(paths, "hack/**")
+	}
+
+	if ct != HelmOperator && ct != Other {
+		paths = append(paths, "apis/grpc/**")
+	}
+
+	if ct == Go {
+		paths = append(paths,
+			"go.mod",
+			"go.sum",
+			"versions/GO_VERSION",
+			"internal/**",
+			"!internal/**/*_test.go",
+			"!internal/db/**",
+			"!internal/k8s/**",
+		)
+	}
+
+	if ct == Rust {
+		paths = append(paths,
+			"rust/Cargo.lock",
+			"rust/Cargo.toml",
+			"rust/bin/agent/**",
+			"rust/libs/ngt-rs/**",
+			"rust/libs/ngt/**",
+			"rust/libs/proto/**",
+			"versions/RUST_VERSION",
+		)
+	}
+
+	if ct == Rust || data.Name == "agent-faiss" {
+		paths = append(paths, "versions/FAISS_VERSION")
+	}
+	if ct == Rust || data.Name == "agent-ngt" {
+		paths = append(paths, "versions/NGT_VERSION")
+	}
```

Comment:
```suggestion
	switch data.ContainerType {
	case DevContainer, CIContainer:
		paths = append(paths,
			"apis/grpc/**"
		        "apis/proto/**"
		        "hack/**",
		)
	case Go:
		paths = append(paths,
		        "apis/grpc/**"
		        "apis/proto/**",
			"go.mod",
			"go.sum",
			"versions/GO_VERSION",
			"internal/**",
			"!internal/**/*_test.go",
			"!internal/db/**",
			"!internal/k8s/**",
		)
	case Rust:
		paths = append(paths,
			"apis/grpc/**"
			"apis/proto/**",
			"rust/Cargo.lock",
			"rust/Cargo.toml",
			"rust/bin/agent/**",
			"rust/libs/ngt-rs/**",
			"rust/libs/ngt/**",
			"rust/libs/proto/**",
			"versions/RUST_VERSION",
			"versions/FAISS_VERSION",
			"versions/NGT_VERSION",
		)
	}
	if strings.EqualFold(data.Name, "agent-faiss") {
		paths = append(paths, "versions/FAISS_VERSION")
	}
	if strings.EqualFold(data.Name, "agent-ngt") {
		paths = append(paths, "versions/NGT_VERSION")
	}
```

Reply:
faiss-agentがrustで書かれていた場合 `FAISS_VERSION` が二回追加されることになってしまうことを懸念して、念の為最後の処理だけ少し変更し、そこ以外はほぼそのまま適用しました！

Reply:
It would be better talk in English
```
Since FAISS_VERSION would be added twice if FAISS-AGENT was written in RUST, we changed only the last process a little and applied it almost as is except there!
```